ern-core/src/FsCache.ts

```diff
@@ -0,0 +1,220 @@
+   * ex: my-miniapp@1.0.0
+   */
+  id: string
+  /**
+   * Absolute path to the entry
+   * ex : /Users/blemair/.ern/cache/78c40fda-d5bb-4681-9bd2-a5fba9df1f70
+   */
+  path: string
+  /**
+   * Size in bytes of the entry
+   */
+  size: number
+  /**
+   * Time (epoc) when the entry was last accessed
+   */
+  lastAccessed: number
+}
+
+const cacheManifestFileName = 'cache-manifest.json'
+export const maxDefaultCacheSize = 2 * 1024 * 1024 * 1024 /* 2GB */
+
+type AddObjectToDirectoryFun<T> = (obj: T, dirPath: string) => Promise<void>
+type ObjectToId<T> = (obj: T) => string
+
+/**
+ * Simple generic file system based LRU cache.
+ * This cache stores arbirtrary objects in unique directories.
+ * The root directory for the cache can be configured, as well as
+ * the maximum size allowed for the cache.
+ * Once cache reaches max size, the least recenctly accessed
+ * objects will be removed from the cache, until size of the cache
+ * falls below maximum size.
+ */
+export class FsCache<T> {
+  /**
+   * Root path to the directory holding the cache
+   */
+  public readonly rootCachePath: string
+  /**
+   * Maximum cache size (in bytes)
+   */
+  public readonly maxCacheSize: number
+  /**
+   * Custom function invoked to add an object entry to the cache
+   */
+  private readonly addObjectToDirectory: AddObjectToDirectoryFun<T>
+  /**
+   * Custom function invoked to convert an object entry to a unique id
+   */
+  private readonly objectToId: ObjectToId<T>
+
+  constructor({
+    /**
+     * Custom function invoked to add an object entry to the cache
+     */
+    addObjectToDirectory,
+    /**
+     * Custom function invoked to convert an object entry to a unique id
+     */
+    objectToId,
+    /**
+     * Maximum cache size (in bytes)
+     */
+    maxCacheSize = maxDefaultCacheSize,
+    /**
+     * Root path to the directory holding the cache
+     */
+    rootCachePath,
+  }: {
+    addObjectToDirectory: AddObjectToDirectoryFun<T>
+    objectToId: ObjectToId<T>
+    maxCacheSize?: number
+    rootCachePath: string
+  }) {
+    if (!addObjectToDirectory) {
+      throw new Error('addObjectToDirectory function is required')
+    }
+    if (!objectToId) {
+      throw new Error('objectToId function is required')
+    }
+    if (!rootCachePath) {
+      throw new Error('rootCachePath is required')
+    }
+    this.rootCachePath = rootCachePath
+    this.maxCacheSize = maxCacheSize
+    this.addObjectToDirectory = addObjectToDirectory
+    this.objectToId = objectToId
+    if (!fs.existsSync(rootCachePath)) {
+      shell.mkdir(rootCachePath)
+    }
+    if (!fs.existsSync(this.cacheManifestPath)) {
+      writeJSONSync(this.cacheManifestPath, { entries: [], size: 0 })
+    }
+  }
+
+  /**
+   * Indicates whether a given object is currently stored in the cache
+   * @param obj The object to check for presence in the cache
+   */
+  public async isInCache(obj: T): Promise<boolean> {
+    const manifestObj = await this.getManifestObj()
```

Comment:
If it's possible for `getManifestObj` to `readJSONSync()` we an can eliminate need for `async`

Reply:
Same as previous comment.